mkdocs/commands/build.py

```diff
@@ -232,75 +233,84 @@ def _build_page(page, config, files, nav, env, dirty=False):
+        # Run `nav` plugin events.
+        nav = config['plugins'].run_event('nav', nav, config=config, files=files)
 
-    # Run `env` plugin events.
-    env = config['plugins'].run_event(
-        'env', env, config=config, files=files
-    )
+        log.debug("Reading markdown pages.")
+        for file in files.documentation_pages():
+            log.debug("Reading: " + file.src_path)
+            _populate_page(file.page, config, files, dirty)
+
+        # Run `env` plugin events.
+        env = config['plugins'].run_event(
+            'env', env, config=config, files=files
+        )
+
+        # Start writing files to site_dir now that all data is gathered. Note that order matters. Files
+        # with lower precedence get written first so that files with higher precedence can overwrite them.
 
-    # Start writing files to site_dir now that all data is gathered. Note that order matters. Files
-    # with lower precedence get written first so that files with higher precedence can overwrite them.
+        log.debug("Copying static assets.")
+        files.copy_static_files(dirty=dirty)
 
-    log.debug("Copying static assets.")
-    files.copy_static_files(dirty=dirty)
+        for template in config['theme'].static_templates:
+            _build_theme_template(template, env, files, config, nav)
 
-    for template in config['theme'].static_templates:
-        _build_theme_template(template, env, files, config, nav)
+        for template in config['extra_templates']:
+            _build_extra_template(template, files, config, nav)
 
-    for template in config['extra_templates']:
-        _build_extra_template(template, files, config, nav)
+        log.debug("Building markdown pages.")
+        for file in files.documentation_pages():
+            _build_page(file.page, config, files, nav, env, dirty)
 
-    log.debug("Building markdown pages.")
-    for file in files.documentation_pages():
-        _build_page(file.page, config, files, nav, env, dirty)
+        # Run `post_build` plugin events.
+        config['plugins'].run_event('post_build', config=config)
 
-    # Run `post_build` plugin events.
-    config['plugins'].run_event('post_build', config=config)
+        if config['strict'] and utils.warning_filter.count:
+            raise SystemExit('\nExited with {} warnings in strict mode.'.format(utils.warning_filter.count))
 
-    if config['strict'] and utils.warning_filter.count:
-        raise SystemExit('\nExited with {} warnings in strict mode.'.format(utils.warning_filter.count))
+        log.info('Documentation built in %.2f seconds', time() - start)
 
-    log.info('Documentation built in %.2f seconds', time() - start)
+    except BuildError as error:
+        # Run `build_error` plugin events.
+        config['plugins'].run_event('build_error', error=error)
+        if error.reraise:
+            raise error
+        raise SystemExit('\nAborted build due to error.')
```

Comment:
The purpose here is to hide the unhelpful (to the user) traceback, not to completely obscure the actual error. We should be passing on the error message to `SystemExit` like we do in the [build command][1].

[1]: https://github.com/mkdocs/mkdocs/blob/master/mkdocs/__main__.py#L153-L155

Reply:
Done!